### DISCUSSION TITLE
Iceberg AWS Glue single catalog limitation

### DISCUSSION THREAD

**Discussion by bgadrian** [trinodb/trino#21972] [Q&A]
Hello,

I am trying to setup multiple Trino catalogs, with same schemas/tables (multitenant scenario) but I noticed that when using AWS Glue Iceberg as metastore, the AWS Glue Database is equivalent with the Schema. 

I know that AWS Glue only restricts to 1 Glue catalog per AWS account/region tuple ( https://docs.aws.amazon.com/glue/latest/dg/components-overview.html#data-catalog-intro ) 

Question: how can I add some sort of prefix/suffix in the Glue catalog and the value should be the Trino catalog, to support my scenario of having multiple Trino catalogs using same Glue catalog

Using docker trino 445 (latest)

Thanks.

**Comment by hashhar** [comment]
It's not possible to do this out of the box.

You'll somehow need to propagate information about what "tenant" is being referred to for a given query and then in https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java you might be able to change the object name which is being referred to.

However this won't be portable at all since those tables couldn't be queried from another system which uses the same Glue without rewriting the queries.

It'd be much much simpler if you just adjust the schema and table names on your end or use a catalog other than Glue.
